libs/utils/android/workloads/exoplayer.py

```diff
@@ -0,0 +1,184 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+
+from subprocess import Popen, PIPE
+
+from time import sleep
+
+from android import Screen, System, Workload
+
+# Regexps for benchmark synchronization
+
+REGEXPS = {
+    'start'    : '.*Displayed com.google.android.exoplayer2.demo/.PlayerActivity',
+    'duration' : '.*period \[(?P<duration>[0-9]+.*)\]',
+    'end'      : '.*state \[.+, .+, E\]'
+}
+
+class Exoplayer(Workload):
+    """
+    Android Exoplayer workload
+    """
+
+    # Package required by this workload
+    package = 'com.google.android.exoplayer2.demo'
+
+    def __init__(self, test_env):
+        super(Exoplayer, self).__init__(test_env)
+        self._log = logging.getLogger('Exoplayer')
+        self._log.debug('Workload created')
+
+    def _play(self):
+
+        # Grant app all permissions
+        System.grant_permissions(self._target, self.package)
```

Comment:
Maybe we should just have the default install method do this?

Reply:
You mean devlib's install ? Afaik this has to be done every time before launching the app if we clear the app data (`System.force_stop(self._target, self.package, clear=True)`), which we want to do to make sure the app doesn't try to launch a previously-run media file.

Reply:
Ah yep makes sense, OK.